src/components/Cooking.js

```diff
@@ -0,0 +1,45 @@
+import Form from "./Form";
+import RecipeBook from "./RecipeBook";
+import { useState } from "react";
+
+const Cooking = () => {
+    const [newRecipes, setNewRecipes] = useState([]);
+    const [userInput, setUserInput] = useState();
+
+    async function api () {
+        const url = new URL('https://api.spoonacular.com/recipes/findByIngredients');
+        url.search = new URLSearchParams ({
+            // apiKey: "ce490440f832431a9cdb9e690f75de8b",
+            // apiKey: "ae04a816ca574350af40a091c219b089",
+            // apiKey: "aad01ceb54a9479592f55feb0f1502b3",
+            apiKey: "667f3f639407497bb32d9f02c53752bc",
+            ingredients: userInput,
+            limitLicense: true,
+            number: 9
+        })
+        await fetch(url)
+            .then((response) => {
+                return response.json();
+            })
+            .then((data) => {
+                setNewRecipes(data);
+            })
+
+        };
+
+    const addIngredient = (event, enteredIngredient) => {
+        event.preventDefault();
+        setUserInput(enteredIngredient);
+        api();   
```

Comment:
Since `api()` makes use of the value of `userInput`, it might make sense to move it into a side effect. When you update a state variable, there's no guarantee that the value will update immediately, so the api might be getting called with the _previous_ value.

To test this, you could type in "apple", click search once. Then, type "orange", and click search again. If I'm right, you'll see apple results come up (until you click a third time, with the ingredient still typed in as "orange")